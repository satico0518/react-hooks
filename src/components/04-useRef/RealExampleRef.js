import React, { useState } from 'react'
import '../counter.css'
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks'

const RealExampleRef = () => {
    const [showComponent, setShowComponent] = useState(true)
    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr/>
            {showComponent && <MultipleCustomHooks />}
            <button className={`btn ${showComponent ? 'btn-danger' : 'btn-success'}`} onClick={() => setShowComponent(!showComponent)}>Toogle Component</button>
        </div>
    )
}

export default RealExampleRef
