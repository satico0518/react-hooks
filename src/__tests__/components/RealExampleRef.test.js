const { shallow } = require('enzyme');
import React from 'react';

import '@testing-library/jest-dom';
import RealExampleRef from '../../components/04-useRef/RealExampleRef';

describe('RealExampleRef', () => {
  test('should match snapdhot', () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should render MultipleCustomHooks', () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper.find('MultipleCustomHooks').exists()).toBeTruthy();
  });
  test('should not render MultipleCustomHooks', () => {
    const wrapper = shallow(<RealExampleRef />);
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBeFalsy();
  });
});
