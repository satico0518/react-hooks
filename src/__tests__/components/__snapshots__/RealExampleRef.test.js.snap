// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef should match snapdhot 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Toogle Component
  </button>
</div>
`;
